hyp3_sdk/hyp3.py

```diff
@@ -20,16 +21,25 @@
 class HyP3:
     """A python wrapper around the HyP3 API"""
 
-    def __init__(self, api_url: str = HYP3_PROD, username: Optional = None, password: Optional = None):
+    def __init__(self, api_url: str = HYP3_PROD, username: Optional = None, password: Optional = None,
+                 prompt: bool = False):
         """
         Args:
             api_url: Address of the HyP3 API
-            username: Username for authenticating to urs.earthdata.nasa.gov.
+            username: Username for authenticating to `urs.earthdata.nasa.gov`.
                 Both username and password must be provided if either is provided.
-            password: Password for authenticating to urs.earthdata.nasa.gov.
+            password: Password for authenticating to `urs.earthdata.nasa.gov`.
                 Both username and password must be provided if either is provided.
+            prompt: Get *unprovided* username and password for authenticating to
+                `urs.earthdata.nasa.gov` from user input
```

Comment:
didn't quite parse `unprovided` until I read the implementation, maybe `Prompt for username and/or password interactively when not provided as keyword parameters`